src/lib/tests/tests.cpp

```diff
@@ -1,6 +1,399 @@
+#include "simd/simd_doubles.h"
+#include "simd/simd_floats.h"
+#include "simd_config.h"
 #include "gtest/gtest.h"
 
-// a blank test
-TEST(DistopiaTests, BlankTest) {
-  SUCCEED();
+/* Section for testing SimdFloatX4 packed structs
+Notes
+* operates on __m128 types which require 16 bit alignment
+
+*/
+
```

Comment:
The swizzles really need testing. They are by far the lowest-level and mistake-prone code in this PR.

Reply:
You already have the swizzles done so will build on top of those.